api/user.go

```diff
@@ -1106,6 +1108,16 @@ func loginWithSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if len(extensionId) != 0 {
+		relayProps["extension_id"] = extensionId
```

Comment:
I can see how if the user initiates a login flow with SAML this validates the extension, but what about other authentication flows? I can understand that you are focus on this auth method but I do believe this should be done for every method in any case